update_offenders.py

```diff
@@ -92,13 +92,10 @@ def write_domains(offenders, existing_domains, mode='wb'):
     }
 
     for o in offenders:
-        existing_domain = existing_domains_dict.get(o.get('domain'))
-        if not existing_domain:
+        domain = o.get('domain')
+        if domain not in existing_domains_dict:
             new_domains.append(o)
```

Comment:
I would just use the "existing_domains_dict" as a based that would be updated if exists, and in the end it will be write it all at once.